nydusify: add layer push retry for copy subcommand

imeoer opened the pull request:
## Relevant Issue (if applicable)
Add push retry mechanism, enhance the success rate for image copy
when a single layer copy failure.

## Details
_Please describe the details of PullRequest._

## Types of changes

_What types of changes does your PullRequest introduce? Put an `x` in all the boxes that apply:_

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Go over all the following points, and put an `x` in all the boxes that apply._

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

imeoer committed:
nydusify: fix layer push retry for copy subcommand

Add push retry mechanism, enhance the success rate for image copy
when a single layer copy failure.

Signed-off-by: Yan Song <yansong.ys@antgroup.com>

liubin commented on contrib/nydusify/pkg/copier/copier.go:
```diff
@@ -66,6 +66,23 @@ type output struct {
+			return nil
+		}
+
+		if total > 0 && !errors.Is(err, context.Canceled) {
```
Is there any other errors that need retry?

imeoer committed:
misc: remove goproxy.io for go build

The goproxy.io service is unstable for now, it effects,
the github CI, let's remove it.

Signed-off-by: Yan Song <yansong.ys@antgroup.com>

BraveY commented on contrib/nydusify/pkg/backend/oss.go:
```diff
@@ -259,6 +260,21 @@ func (b *OSSBackend) Type() Type {
+
+func (rr *RangeReader) Reader(offset int64, size int64) (io.ReadCloser, error) {
+	rc, err := rr.b.bucket.GetObject(rr.blobID, oss.Range(offset, offset+size-1))
+	return rc, err
```
how about `return rr.b.bucket.GetObject(rr.blobID, oss.Range(offset, offset+size-1))`?

BraveY reviewed (approved):
LGTM.
